format/Schema.fbs

```diff
@@ -187,6 +187,103 @@ table Interval {
   unit: IntervalUnit;
 }
 
+
+/// ----------------------------------------------------------------------
+/// user defined key value pairs to add custom metadata to arrow
+/// key namespacing is the responsibility of the user
+
+table KeyValue {
+  key: string;
+  value: string;
+}
+
+/// ------------------------------------------------------------------------
+/// Types used to represent a packed c-like structs.  These are experimental
+/// and maybe subject to change.
+/// IMPORTANT: COMPATIBILITY MAY BE BROKEN IN FUTURE release.
+/// They are not required to be implemented.
+
+/// Similar to a c-union, takes the byte-width
+/// of its longest sub-type.  To record type of union
+/// use the "tagged union" pattern.
+table PackedUnion {
+}
+
+/// A binary string of a maximum size.  Fields are layed out as
+/// [length][UTF8 bytes].  The number of bytes used is maxSize +
+/// lengthType.bitWidth/8.
+table PackedBinary {
+  /// Maximum number of bytes (not characters) used.  This
+  /// amount of bytes will always be used.
+  maxSize: int;
+
+  /// If not set all binary data are of maxSize.  If set use specified
+  /// numeric type to specify a length less then or equal to
+  /// max size.
+  lengthType: Int;
```

Comment:
Hmm... how does an int denote a numeric type? What is the encoding?

Reply:
This is the same Int type used above.  I was imaging same encoding as the rest of the the values (governed by endianness at a higher level).  I will add some more documentation here.

Reply:
More precisely: this field encodes a _type_, right? (it's called `lengthType`, not `length`). My question is how that type is encoded.